### DISCUSSION TITLE
Stop Shoving Github Mobile Down Our Throats as 2FA/Authenticator Choice

### DISCUSSION THREAD

**Discussion by FlorianWendelborn** [community/community#31942] [Mobile] [answered]
I don’t remember ever being asked if I want to use GitHub mobile as my 2FA device. It is severely annoying and unexpected that GitHub **ALWAYS** tries to shove it down my throat and makes me click extra just to be able to enter the normal, standardized 2FA/TOTP token.

There doesn’t seem to be any way of turning that feature off, besides stopping to use GitHub mobile. The app is unfortunately very good so I don’t want to stop using it, but that doesn’t mean that I’d EVER want it to be my 2FA device.

Please stop this disgusting and insecure (since unexpected!) marketing practice or at least give us a way to opt-out from it.

The GitHub settings even lie about this explicitly:

![image](https://user-images.githubusercontent.com/1133858/188322124-92900660-eb29-4a1e-94e0-971510a80605.png)

No, GitHub, currently my primary 2FA device seems to be GitHub’s mobile app, not by choice, but because of this terrible system that only exists for marketing.

**Comment by candyho** [comment, marked as answer]
Hey, thanks for your feedback here. I hear your frustration about how multi-factor authentication defaults to GitHub Mobile when you have other factors enabled like TOTP.

I've shared your notes with our team internally about giving users more control over GitHub Mobile as an authentication method and prioritizing other multi-factor methods over GitHub Mobile without relying on stored cookies. We'll look to improve on the current implementation to give users more control here.

In the meantime, if you've uninstalled GitHub Mobile, revoking the GitHub Mobile OAuth apps will remove them as a multi-factor method when logging in. You can find revoke either app here:

- [GitHub for Android](https://github.com/settings/connections/applications/3f8b8834a91f0caad392)
- [GitHub for iOS](https://github.com/settings/connections/applications/2cfa9b7a1b57de32dd0d)

Thanks for helping us improve the security experience here at GitHub!

**Reply by FlorianWendelborn** [reply]
Thanks for looking into this and for a professional response. I just encountered this annoyance for the hundredth time before I wrote this, so sorry for conveying the frustration in this post 😅

**Comment by eliperkins** [comment]
Thanks for your feedback, y'all! Today, we've shipped [a new feature that allows for setting a preferred option for two-factor authentication](https://github.blog/changelog/2023-02-22-preferred-2fa-methods-and-settings-improvements/). This will allow you to set other two-factor authentication methods to be used before another option. [Go to Settings > Password and Authentication on GitHub.com](https://github.com/settings/security) in your browser to set a different preferred two-factor option.

<img width="772" alt="Screenshot 2023-02-22 at 4 46 23 PM" src="https://user-images.githubusercontent.com/1051453/220766920-fc0cacc4-5539-4afc-9940-83acbd17efee.png">

Additionally, [GitHub Mobile sessions can be revoked within Settings](https://github.com/settings/sessions?#mobile_sessions) as well:

<img width="793" alt="Screenshot 2023-02-22 at 4 49 54 PM" src="https://user-images.githubusercontent.com/1051453/220767378-9bdca298-994d-4cb8-8b7c-22724b16e2a1.png">

**Reply by FlorianWendelborn** [reply]
Yeah, I noticed it’s a thing now. MUCH better <3

**Reply by jplatte** [reply]
It's an improvement in terms of annoyance, but not in terms of security. If one's regular 2FA method is more secure than the app (not that hard unless it's SMS...), you still have to choose between

- using the app and
- having a well-secured 2FA setup

because it's impossible to use the app, but disallow its use for 2FA.

**Reply by FlorianWendelborn** [reply]
@jplatte Given that this issue is already marked as solved I think you should open a new issue and link this one if you think it's related. Make sure you specify it's security related
